### PR TITLE
aten.pow: Fix integer argument accuracy

### PR DESCRIPTION
**Pull request by mgehre-amd**
Fixes pow(float,int) and pow(int,float) accuracy.
Torch uses f64 internally to compute pow if one argument is integer.
Fixes accuracy issues seen on ONNX.

The `onnx.Pow(int,int)` case still produces numeric mismatches for values that overflow.

### CODE REVIEW

File: test/Conversion/TorchOnnxToTorch/simple_ops_g_to_p.mlir [hunk 1]
```diff
@@ -1102,12 +1102,9 @@ func.func @test_pow(%arg0: !torch.vtensor<[3,4,5],f32>, %arg1: !torch.vtensor<[3
 func.func @test_pow_i32(%arg0: !torch.vtensor<[3,4,5],si32>, %arg1: !torch.vtensor<[3,4,5],si32>) -> !torch.vtensor<[3,4,5],si32> attributes {torch.onnx_meta.ir_version = 8 : si64, torch.onnx_meta.opset_version = 15 : si64, torch.onnx_meta.producer_name = "backend-test", torch.onnx_meta.producer_version = ""} {
   // CHECK: %[[FALSE:.+]] = torch.constant.bool false
   // CHECK: %[[NONE:.+]] = torch.constant.none
-  // CHECK: %[[DTY:.+]] = torch.constant.int 6
-  // CHECK: %[[CAST_LHS:.+]] = torch.aten.to.dtype %arg0, %[[DTY]], %[[FALSE]], %[[FALSE]], %[[NONE]]
-  // CHECK: %[[CAST_RHS:.+]] = torch.aten.to.dtype %arg1, %[[DTY]], %[[FALSE]], %[[FALSE]], %[[NONE]]
-  // CHECK: %[[POW:.+]] = torch.aten.pow.Tensor_Tensor %[[CAST_LHS]], %[[CAST_RHS]]
+  // CHECK: %[[POW:.+]] =  torch.aten.pow.Tensor_Tensor %arg0, %arg1 : !torch.vtensor<[3,4,5],si32>, !torch.vtensor<[3,4,5],si32> -> !torch.vtensor<[3,4,5],f64>
```
Comment (mgehre-amd) [inline test/Conversion/TorchOnnxToTorch/simple_ops_g_to_p.mlir:1105]: This isn't quite the correct lowering, but it works for most numbers. 
The correct lowering would go to `torch.aten.pow.Tensor_Tensor si32,si32 -> si32`,
but that op cannot be lowered to linalg.
As I didn't want to break existing support, I kept it like that.

Comment (TinaAMD) [reply test/Conversion/TorchOnnxToTorch/simple_ops_g_to_p.mlir:1105]: Ok, to summarize the changes here: Before int/float, float/int and int/int was not working properly, and after the changes int/float combinations will work (accuracy is now fixed by using `f64`) and the int/int case is still not properly implemented?

Comment (mgehre-amd) [reply test/Conversion/TorchOnnxToTorch/simple_ops_g_to_p.mlir:1105]: correct. The `int/int` case works the same as before this PR: It produces correct results when no overflow occurs, and incorrect results on overflows.

**Review by TinaAMD — APPROVED** [review]
LGTM

**Pull request merged**